hphp/runtime/vm/jit/vasm-util.cpp

```diff
@@ -29,7 +29,12 @@ bool is_nop(const copy& i) { return i.s == i.d; }
 bool is_nop(const copy2& i) { return i.s0 == i.d0 && i.s1 == i.d1; }
 
 // movb r,r is a nop, however movl is not since it zeros upper bits.
-bool is_nop(const movb& i) { return i.s == i.d; }
+bool is_nop(const movb& i) {
+  // On PPC64 movb r,r cannot be considered a trivial nop since it zeros
```

Comment:
I'm not sure if this is the case, but remember that in PPC64 there is no register of 8 bits, so if it's moving a value into a 8 bit register, later this register would be used as 8 bit register. The issue is that if I move a 8 bit value to a 8 bit register, and later use this register in any other instruction, on PPC64 it will consider the entire register, considering the garbage of the higher bits.
@racardoso check if this is the case.